utils/utils.mk

```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 ETH Zurich and University of Bologna.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Sergio Mazzola <smazzola@iis.ee.ethz.ch>
+
+ifndef utils_mk
+utils_mk=1
+
+CHIM_UTILS_DIR ?= $(CHIM_ROOT)/utils
+
+# Download padrick binary
+$(CHIM_UTILS_DIR)/padrick:
+	cd $(CHIM_UTILS_DIR) && curl https://api.github.com/repos/pulp-platform/padrick/releases/tags/v0.3.6 \
+	| grep "browser_download_url.*Padrick-x86_64.AppImage" \
+	| cut -d : -f 2,3 \
+	| tr -d \" \
+	| wget -qi -
+	mv $(CHIM_UTILS_DIR)/Padrick-x86_64.AppImage $(CHIM_UTILS_DIR)/padrick
+	chmod a+x $(CHIM_UTILS_DIR)/padrick
```

Comment:
This seems very shaky :/

Reply:
Looks like this is what Manuel implemented also for other repos: https://github.com/pulp-platform/pulpissimo/blob/master/utils/utils.mk#L32-L42

Reply:
Exactly, that's where I took inspiration for, also for the include guards. I will make them more specific.
Now that I check, it seems like we can also install padrick through pip. I can change the flow to that.
For the gitignore, my ratio was that if the makefile are directory-specific, than each directory should be responsible of letting git know what to ignore (based on the artifacts produced by the makefile, e.g. the padrick binary in utils, etc...). But I don't have a strong opinion, I can collapse everything on the root gitignore.

Reply:
Yeah, imo distributed gitignore files get complicated to keep track of, as rules might enable and disable each other in various levels. I'm strongly in favour of a central ruleset.

If padrick works with pip that would be awesome, otherwise I guess we can use a binary. This install script seems very complicated though, and hardcoding `x86_64` ISA seems prone to failure.